src/test/kotlin/com/exactpro/th2/common/schema/grpc/router/impl/DefaultGrpcRouterTest.kt

```diff
@@ -0,0 +1,802 @@
+                        Int.MAX_VALUE,
+                        RETRY_TIMEOUT,
+                        RETRY_TIMEOUT,
+                        retryInterruptedTransaction
+                    )
+                )
+            )
+            return grpcRouterClient.getService(TestService::class.java)
+        }
+
+        protected fun createClientAsync(
+            configuration: GrpcServiceConfiguration = GrpcServiceConfiguration(
+                RobinRoutingStrategy().apply { init(GrpcRawRobinStrategy(listOf("endpoint"))) },
+                TestService::class.java,
+                mapOf("endpoint" to GrpcEndpointConfiguration("localhost", SERVER_PORT))
+            ),
+            retryInterruptedTransaction: Boolean = false
+        ): AsyncTestService {
+            grpcRouterClient.init(
+                GrpcConfiguration(services = mapOf("test" to configuration)),
+                GrpcRouterConfiguration(
+                    retryConfiguration = GrpcRetryConfiguration(
+                        Int.MAX_VALUE,
+                        RETRY_TIMEOUT,
+                        RETRY_TIMEOUT,
+                        retryInterruptedTransaction
+                    )
+                )
+            )
+            return grpcRouterClient.getService(AsyncTestService::class.java)
+        }
+
+        protected fun createServer(
+            configuration: GrpcServerConfiguration = GrpcServerConfiguration(
+                null,
+                SERVER_PORT
+            ),
+            grpcRouter: DefaultGrpcRouter = grpcRouterServer,
+            completeResponse: Boolean = true,
+            handlerBaton: Baton? = null,
+        ): Server {
+            grpcRouter.init(GrpcConfiguration(serverConfiguration = configuration), GrpcRouterConfiguration())
+            return grpcRouter.startServer(TestServiceHandler(completeResponse, handlerBaton)).apply(Server::start)
+        }
+    }
+
+    @Nested
+    inner class SingleRequestSingleResponseSyncTest : AbstractGrpcRouterTest() {
+        @Test
+        override fun general() {
+            createServer().execAndClose(true) {
+                val response = executor.submit<Response> {
+                    return@submit createClientSync().singleRequestSingleResponse(Request.newBuilder().setSeq(1).build())
+                }.get(1, TimeUnit.MINUTES)
+
+                assertEquals(1, response.seq)
+                assertEquals(1, response.origSeq)
+            }
+        }
+        @Test
+        override fun `delayed server start`() {
+            val clientServerBaton = Baton("client-server")
+            val future = executor.submit<Response> {
+                clientServerBaton.give("client thread started")
+                return@submit createClientSync().singleRequestSingleResponse(Request.newBuilder().setSeq(1).build())
+            }
+
+            clientServerBaton.get("wait client thread start")
+            Thread.sleep(RETRY_TIMEOUT * 2)
```

Comment:
Why do we need to wait here? If you need to make sure the task is executed within a timeout you can use `Assertions.assertTimeoutPreemptively` method

Reply:
I use fixed retry timeout in these tests. I wait time to grantee that client switch to retry mode. The time can be reduced because gRPC router begin retry after first failure